src/rust/engine/process_execution/src/local.rs

```diff
@@ -21,6 +23,53 @@ impl CommandRunner {
   pub fn new(store: fs::Store, fs_pool: Arc<fs::ResettablePool>) -> CommandRunner {
     CommandRunner { store, fs_pool }
   }
+
+  fn construct_output_snapshot(
+    store: Store,
+    posix_fs: Arc<fs::PosixFS>,
+    output_file_paths: BTreeSet<PathBuf>,
+    output_dir_paths: BTreeSet<PathBuf>,
+  ) -> BoxFuture<Snapshot, String> {
+    let output_dirs_glob_strings: Result<Vec<String>, String> = output_dir_paths
+      .into_iter()
+      .map(|p| {
+        p.into_os_string()
+          .into_string()
+          .map_err(|e| format!("Error stringifying output_directories: {:?}", e))
+          .map(|s| format!("{}/**", s))
+      })
+      .collect();
+
+    let output_dirs_future = posix_fs
+      .expand(
+        PathGlobs::create(
+          &try_future!(output_dirs_glob_strings),
+          &[],
+          StrictGlobMatching::Ignore,
+        ).unwrap(),
```

Comment:
Either comment that this `unwrap()` is infallible (because of how we're constructing the `PathGlobs`) or (probably preferred) wrap the `PathGlobs::create` in a `try_future!` to handle its error

Reply:
fixed!